crates/client/settlement/src/errors.rs

```diff
@@ -0,0 +1,63 @@
+use std::time::Duration;
+
+use async_trait::async_trait;
+use futures_timer::Delay;
+
+use crate::RetryStrategy;
+
+/// Settlement error type.
+#[derive(thiserror::Error, Debug)]
+#[allow(missing_docs)]
+pub enum Error {
+    #[error("Blockchain error: {0}")]
+    Blockchain(#[from] sp_blockchain::Error),
+
+    #[error("Starknet API error: {0}")]
+    StarknetApi(#[from] starknet_api::StarknetApiError),
+
+    #[error("Failed to find Madara log: {0}")]
+    DigestLog(#[from] mp_digest_log::FindLogError),
+
+    #[error("Failed to find Substrate block hash for Starknet level: {0}")]
+    MappingSync(u64),
+
+    #[error("Failed to find Substrate block header for hash: {0}")]
+    UnknownBlock(String),
```

Comment:
It's `<B as Block>::Hash`, but we would need to make Error generic then

Reply:
Sound ok for me to make it generic. As this error is clearly linked to the substrate framework

Reply:
Not for the particular settlement provider implementations, e.g. for Ethereum contract

Reply:
The error message do mention Substrate: `    #[error("Failed to find Substrate block header for hash: {0}")]`
So I assume it will always be Substrate related